LEDGAME.ino

```diff
@@ -20,181 +23,187 @@ void setup()
       } 
 }
 
-
 class Timer
 {
   public:
-  unsigned long  i_timer;
-  unsigned long  i_ledTimer;
-  unsigned long  i_oldLedTimer;
+  unsigned long  newTimeValue;
   unsigned long  limitValue;
-  unsigned long old_i_timer;
+  unsigned long  oldTimeValue;
 
   void updateTimer()
   {
-     i_timer = millis();
+     newTimeValue = millis();
   }
   
-  void setTimer(unsigned long timeToSet)
-  {
-    old_i_timer = millis();
-    limitValue = timeToSet;
-  }
+    void setTimer(unsigned long timeToSet)
+    {
+      oldTimeValue = millis();
+      limitValue = timeToSet;
+    }
   
-  bool isTimerRun()
-  {
-    if ( (i_timer - old_i_timer) <= limitValue)
-      return true;
-    else 
-      return false;
-  }
+      bool isTimerRunning()
+      {
+        newTimeValue = millis();
+        
+        if ( (newTimeValue - oldTimeValue) <= limitValue)
+          return true;
+        else 
+          return false;
+      }
 
-  //таймер лампочки 3сек
-  void Led_timer() 
-  {
-     buttWait(); // it is not right to do so - LedTImer mus be a led timer - it do not have to deal with buttons. You better run buttonWait in the loop()
-  }
 };
 
 Timer timer; 
 Timer ledTimer;
+Timer waitTimer;
 
-bool isLedTimerRun()
-{
-  unsigned long  newTimeLed;
-  newTimeLed = millis();
-  if ( (newTimeLed - timer.i_oldLedTimer) >= ledTimer.limitValue )
-  {
-    return false;
-  }
-  else
-  {
-    return true;
-  }
-  
-}
 
-//ожидаем нажатие кнопки 
-int buttPress(int num)
-{
-  noButtPress = 0;
-  timer.i_oldLedTimer = millis();
-   
-  while (butt[num] == 0)
-  {
 
-  if (digitalRead(buttInit[num]) == HIGH) {
-    butt[num] = num;
-    break;
-    } 
-    
-   else if ( isLedTimerRun() == false )
-    {
-       noButtPress = 1;
-     break;
-    }
-  }
-    return 0;
-  }
 
 
-int randomgen()
+//generating random number for led
+int randomGenLed()
 {
   int num = rand() % 3 + 1; 
    return num; 
 }
 
-//включение рандомной лампочки
-int random_led(int num)
+//generating random time for led
+int randomGenTime()
 {
-   digitalWrite(ledInit[num], HIGH);   
+  int randomtime = (rand() % 9 + 3) * 100; 
+  return randomtime;
 }
 
+ //turn on random led
+int randomLedOn(int num)
+{
+   digitalWrite(ledInit[num], HIGH);   
+}
 
 
-//выключаю все кнопки 
-void led_off()
+//turn off all leds
+void ledOff(int num)
   {
-    // here you also can use a cycle
-   digitalWrite(ledInit[1], LOW);
-   digitalWrite(ledInit[2], LOW);
-   digitalWrite(ledInit[3], LOW);
-   digitalWrite(ledInit[0], LOW);
+    digitalWrite(ledInit[num], LOW);
   }  
 
-//выдаем результат в зависимости от нажатой кнопки 
-int buttWait(int num)
+//буфер с нажатыми кнопками
+int WhichButtPressed()

```

Comment:
un-correct name of the function: this function does not return anything and it just sets but[c] = c; if any button is pressed.
Maybe it is better to call it checkAnyButtonPressed() ???
and think about the buttonOldState and buttonNewState